noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -151,6 +152,34 @@ contract NFT {
         let to_note_slot = storage.private_nfts.at(to).storage_slot;
         let hiding_point = NFTNote::hiding_point().new(to_npk_m_hash, note_randomness, to_note_slot);
 
+        // ####### HACKY PARTIAL NOTE EMISSION CODE (to be replaced in a followup PR with proper macro support) #######
+        // TODO(#8769): The following is a temporary partial notes delivery solution that should be eventually replaced.
+        let to_public_keys = get_public_keys(to);
+        let hacky_nft_note = crate::hacky_partial_note::NFTNote {
+            npk_m_hash: to_npk_m_hash,
+            randomness: note_randomness,
+            header: dep::aztec::prelude::NoteHeader::empty()
+        };
+        // `from` cannot receive any logs here as that would leak information of who is the note recipient. For this
+        // reason we encrypt outgoing to `to`.
+        let encrypted_log: [u8; 480] = dep::aztec::encrypted_logs::payload::compute_encrypted_note_log(
+            context.this_address(),
+            to_note_slot,
+            context.request_ovsk_app(to_public_keys.ovpk_m.hash()),
+            to_public_keys.ovpk_m,
+            to_public_keys.ivpk_m,
+            to,
+            hacky_nft_note
+        );
+        let log_hash = dep::aztec::protocol_types::hash::sha256_to_field(encrypted_log);
+
+        // Unfortunately we need to push a dummy note hash to the context here because a note log requires having
+        // a counter that corresponds to a note hash in the same call.
+        let note_hash_counter = context.side_effect_counter;
+        context.push_note_hash(5);
```

Comment:
is this the note content hash or the actual note hash (with the nonce)? because if former, wouldn't it fail the second time we call this?

Reply:
It's not, it's just a dummy note hash because a note log requires having a counter that corresponds to a note hash in the same call.